test/libsolidity/syntaxTests/specialFunctions/abidecode/abi_decode_invalid_argument_tuple_type.sol

```diff
@@ -0,0 +1,9 @@
+contract C {
+    function f() pure public {
+        abi.decode("", ((uint))); // This was accepted before
```

Comment:
This is accepted by the current (and past, I assume) versions of the compiler. I guess it is evaluated in the end as just `uint` instead of a unitary tuple.
Should this PR continue to accept unitary nested tuples as before?

Reply:
I would disallow them if it can be done without too much complication. It would be in line with the goal of the PR since it's an expression, not just a type name. Just make sure that `abi.decode(..., (uint))` keeps working. Disallowing that would make decoding some things impossible.

Unitary tuples are weird in Solidity. Technically, `(x)` should always be seen as an expression equivalent to `x`. But I think we have some special casing so that this is actually interpreted as a tuple in places where we expect tuples. So I suspect that currently `((uint))` currently reduces to `(uint)`, not `uint` in this particular context and that's why it works. I.e. the outer parenthesis is probably treated specially.

Reply:
> I would disallow them if it can be done without too much complication. It would be in line with the goal of the PR since it's an expression, not just a type name.

I agree. The changes introduced by this PR do disallow the unitary tuple.
I was just in doubt because before the changes the compiler accepts it.